### PR TITLE
Make MetricsFeature's dependencies explicit

### PR DESCRIPTION
**Pull request by goedderz**
### Scope & Purpose

Make the dependencies of MetricsFeature explicit, so it can get by without getFeature calls, so it can be an ApplicationFeature instead of an ArangodFeature, so it can be used separately from it.

In order to achieve that, this PR introduces LazyApplicationFeatureReference to handle potential circular references in the feature dependencies (which we should try to get rid of later, whenever possible).

- [X] :hammer: Refactoring/simplification

#### Related Information

- [X] Enterprise PR: https://github.com/arangodb/enterprise/pull/1492
- [X] Jira ticket: https://arangodb.atlassian.net/jira/software/c/projects/TD/issues/TD-1

### CODE REVIEW

File: arangod/Metrics/MetricsFeature.cpp [hunk 1]
```diff
@@ -365,4 +388,12 @@ void MetricsFeature::batchRemove(std::string_view name,
...
 }
 
+void MetricsFeature::prepare() {
+  _queryRegistryFeature = std::move(_lazyQueryRegistryFeatureRef).get();
+  _statisticsFeature = std::move(_lazyStatisticsFeatureRef).get();
+  _engineSelectorFeature = std::move(_lazyEngineSelectorFeatureRef).get();
+  _clusterMetricsFeature = std::move(_lazyClusterMetricsFeatureRef).get();
+  _clusterFeature = std::move(_lazyClusterFeatureRef).get();
+}
+
```
Comment (alexbakharew) [inline arangod/Metrics/MetricsFeature.cpp:398]: Should we assert somehow that `prepare()` and therefore `get()` methods are really called only once?

**Review by jsteemann — APPROVED** [review]
LGTM

File: arangod/Metrics/MetricsFeature.cpp (hunk 1 above)
Comment (goedderz) [reply arangod/Metrics/MetricsFeature.cpp:398]: It is asserted that `get()` is called only once: https://github.com/arangodb/arangodb/pull/20945/files?show-viewed-files=true&file-filters%5B%5D=#diff-ba7c32ee3716b0719387545111a837bd0a213f052957a14db2d32b9ee24be3ccR103

**Review by alexbakharew — APPROVED** [review]
LGTM

**Pull request merged**